### PR TITLE
Use Linear Programming to initialize potencies

### PR DESCRIPTION
**Pull request by Malleek**

### CODE REVIEW

File: include/PetriEngine/PQL/Contexts.h [hunk 1]
```diff
@@ -235,6 +236,9 @@ namespace PetriEngine {
 
             glp_prob* makeBaseLP() const;
 
+            mutable std::vector<uint32_t> xs;
```
Comment (petergjoel) [inline include/PetriEngine/PQL/Contexts.h:239]: `xs` seems like a funky naming scheme? Also please use the `_` prefix for fields.

File: src/VerifyPN.cpp [hunk 2]
```diff
@@ -671,3 +671,138 @@ void simplify_queries(  const MarkVal* marking,
...
+                    if (options.queryReductionTimeout > 0 && qt > 0) {
+                        SimplificationContext simplificationContext(marking, net, qt,
+                            options.lpsolveTimeout, &cache, true);
+                        try {
+                            negstat_t stats;
+                            auto simp_cond = PetriEngine::PQL::simplify(queries[i], simplificationContext);
+                            // queries[i] = pushNegation(simp_cond.formula, stats, context, false, false, true);
+                            wasAGCPNApprox |= dynamic_cast<NotCondition*> (queries[i].get()) != nullptr;
+
+                            potencies = simplificationContext.xs;
```
Comment (petergjoel) [inline src/VerifyPN.cpp:748]: What happens when we have multiple queries?

File: src/VerifyPN.cpp [hunk 3]
```diff
@@ -671,3 +671,138 @@ void simplify_queries(  const MarkVal* marking,
             return a;
     }) && std::chrono::duration_cast<std::chrono::seconds>(end - begin).count() < options.queryReductionTimeout && to_handle > 0);
 }
+
+void simplify_queries_potency(const MarkVal* marking,
```
Comment (petergjoel) [inline src/VerifyPN.cpp:675]: Is this code duplicated from somewhere ? It looks like it.

Comment (Malleek) [reply src/VerifyPN.cpp:675]: Yes, it is the function simplify_queries with less code, and with the potencies = [...] line added

Comment (Malleek) [reply src/VerifyPN.cpp:675]: I could add 2 optional parameters to the simplify_queries function (bool isLp and initPotencies), and then do the correct things according to the bool isLP

**Review by petergjoel — APPROVED** [review]

**Comment by srba** [comment]
Are these new options related to lp-initizalization of potencies?
--init-potency <timeout>             Initial potency timeout in seconds (default 10)
  --interval-timeout <timeout>         Time in seconds before the max intervals is halved (default 10)
                                       write --interval-timeout 0 to disable interval limits
If yes, then add that the second one is for potency initialization

Also, make 
--use-lp-potencies                   Use LP for computing the initial potencies
the default option, so that it is run by default and add the switch --do-not-use-lp-potencies 
to disable this LP initialization of potencies.

Finally, please, adjust the competition script accordingly, test it again (rerun the consistency and performance tests) and upload the new script tapaal.sh into 2024 competition scripts.

**Comment by Malleek** [comment]
The new changes are:
- Potency initialization is enabled by default. It only runs when the strategy is RPFS or RandomWalk
- Only one option: --init-potency-timeout <timeout>. Write 0 to disable.
- The new tapaal script is similar to the previous one, but uses --init-potency-timeout option

The consistency and performance tests are currently running.

**Review by srba — APPROVED** [review]

**Pull request merged**